### ISSUE TITLE
Segfault on arm due to sync/atomic

### ISSUE THREAD

**Issue by paralin** [fsouza/go-dockerclient#525]
I am getting this panic on running AddEventListeners:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x4 pc=0xf04d8]

goroutine 1 [running]:
panic(0x465ed0, 0x1090a038)
        /usr/lib/go/src/runtime/panic.go:464 +0x330
sync/atomic.storeUint64(0x1090e98c, 0x0, 0x0)
        /usr/lib/go/src/sync/atomic/64bit_arm.go:20 +0x40
github.com/fsouza/go-dockerclient.(*eventMonitoringState).enableEventMonitoring(0x1090e960, 0x1090e9b0, 0x0, 0x0)
        /home/paralin/Documents/goworkspace/src/github.com/fsouza/go-dockerclient/event.go:165 +0xc0
github.com/fsouza/go-dockerclient.(*Client).AddEventListener(0x1090e9b0, 0x10ac8640, 0x0, 0x0)
        /home/paralin/Documents/goworkspace/src/github.com/fsouza/go-dockerclient/event.go:92 +0x58
```

The panic is in storeUint64 in 64bit_arm atomic, line 165 on event.go in go-dockerclient.

Apparently it's something to do with alignment of structs: 

 - https://github.com/tpng/afero/commit/9132fc2caa2e5cfa428f8665189923b5236c4f0f
 - https://github.com/spf13/hugo/issues/1621

**Comment by paralin** [comment]
https://github.com/paralin/go-dockerclient/commit/898d47a4b0b6b75e476b8091207e128c18969164

This is a good workaround, and works fine on arm (tested).

**Comment by paralin** [comment]
Fixed by #522, please pull

**Comment by fsouza** [comment]
@paralin thanks for reporting and confirming that #522 fixes it!
